routes/v1.js

```diff
@@ -147,10 +147,27 @@ function addRoutes(app, peliasConfig) {
   );
 
   // defer to pelias parser for analysis IF there's no response AND placeholder should not have executed
-  const shouldDeferToPeliasParser = all(
-    not(predicates.hasRequestErrors),
-    not(predicates.hasResponseData)
-  );
+  const shouldDeferToPeliasParser = function(req, res) {
+    // if no results returned from placeholder or previous elasticsearch queries, short circuit and return true
+    // (we always want to try pelias parser based queries if there's no results)
+    if (not(predicates.hasResponseData)(req, res)) {
+      return true;
+    }
+
+    //// All conditions below must be true for us to use pelias parser
+
+    // if the 'sources' parameter is only wof, no need to do anything. Placeholder can return all the possible answers
+    const sources_not_only_wof = not(predicates.isRequestSourcesOnlyWhosOnFirst)(req,res);
+
+    // if there are only admin results, but parse contains more granular items than admin,
+    // then we want to defer to pelias parser based queries
+    const admin_only = all(
+      predicates.hasAdminOnlyResults,
+      not(predicates.isAdminOnlyAnalysis)
+    )(req, res);
+
+    return sources_not_only_wof && admin_only;
```

Comment:
minor nit: are there pros/cons to having `sources_not_only_wof` outside of the `all()` and then using `&&`?

I'm guessing you did it like this so you could document it better with the 👍 comments?

Reply:
Yeah, It was a bit easier to work through the logic. We could rewrite it, perhaps something like this:

```javascript
  const shouldDeferToPeliasParser2 = any(
    // we always want to try pelias parser based queries if there are no results
    not(predicates.hasResponseData),
    all(
      // if there are only admin results, but parse contains more granular items than admin,
      // then we want to defer to pelias parser based queries
      not(predicates.isAdminOnlyAnalysis),
      predicates.hasAdminOnlyResults
      // exception: if the 'sources' parameter is only wof, do not use pelias parser
      // in that case Placeholder can return all the possible answers
      not(predicates.isRequestSourcesOnlyWhosOnFirst),
    )
  );

```

Reply:
After some thought I like the standard predicate version better. So I've gone with that after testing that it does indeed behave the same.